src/sage/combinat/k_regular_sequence.py

```diff
@@ -1212,18 +1263,17 @@ def guess(self, f, n_max=100, max_exponent=10, sequence=None):
             seq = lambda m: vector([])
         else:
             mu = [M.rows() for M in sequence.mu]
-            seq = lambda m: sequence.left * sequence._mu_of_word_(
-                self._n_to_index_(m))
+            seq = lambda m: (sequence._mu_of_word_(self._n_to_index_(m))
+                             * sequence.right)
```

Comment:
This can probably be rewritten as `sequence.coefficient_of_word(self._n_to_index_(m), multiply_left=False)`
in order to use that public method instead of using the private method `_n_to_index_`.

Reply:
`__getitem__` is the thing to use here. However, cannot use `[...]` directly due to the optional keyword parameter. I have now, in analogy to `RecognizableSeries` implemented `coefficient_of_n` as shortcut. (The name is not ideal, but I did not come up with something better and appearently also not earlier as there is `_n_to_index_`.

1c56e7fed36 please review.

Reply:
ok, fine, please resolve.